### PR TITLE
Add events to registry contracts

### PR DESCRIPTION
**Pull request by ewilz**

### CODE REVIEW

File: test/TokenCuratedRegistry.behavior.js [hunk 1]
```diff
@@ -55,26 +55,9 @@ function shouldBehaveLikeTokenCuratedRegistry ({
           await this.registry.setChallenge(itemId, this.challenge.address)
         })
 
-        describe('and the challenge has failed', function () {
-          beforeEach(async function () {
-            await this.challenge.set_mock_passed(false)
-            await this.registry.remove(itemId)
-          })
-
-          it('removes the item', async function () {
```
Comment (ewilz) [inline test/TokenCuratedRegistry.behavior.js:64]: Challenge should be resolved, or else ItemOwner will never get reward, and tokens will be locked in the registry

Comment (mikec) [reply test/TokenCuratedRegistry.behavior.js:64]: nice find! it simplifies this a lot, that was totally an unnecessary require check

File: contracts/Registry/TokenCuratedRegistry.sol [hunk 2]
```diff
@@ -6,6 +6,13 @@ import '../Challenge/IChallengeFactory.sol';
 import '../Challenge/IChallenge.sol';
 
 contract TokenCuratedRegistry is StakedRegistry, TimelockableItemRegistry {
+
+  event Application(bytes32 indexed itemID, address indexed itemOwner, uint appEndDate);
```
Comment (mikec) [inline contracts/Registry/TokenCuratedRegistry.sol]: might want to spell out the var name, `applicationEndDate`

**Comment by mikec** [comment]
👍

**Pull request merged**